torch/_inductor/compile_fx.py

```diff
@@ -587,6 +587,11 @@ def _shape_env_from_inputs(inputs):
     if fake_mode is not None:
         return fake_mode.shape_env
 
+    # When there are no tensor inputs, get shape_env from the first SymInt.
+    for input in inputs:
+        if isinstance(input, torch.SymInt):
```

Comment:
Why only integers?
```suggestion
        if isinstance(input, (torch.SymInt, torch.SymFloat, torch.SymBool)):
```

Reply:
Yes, I've left the comment above. Will change that and add more tests.

Reply:
So, in inductor, SymInts only will be fine as we are not supporting non-int inputs see https://docs.google.com/document/d/1INSCdYu1PxXcr43HrD82OudeEuS-qxQe1yZmLg2wy6A/edit#heading=h.aytveihghlgn